src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -27,27 +32,61 @@ public class PaymentGatewayClient {
     PostePayInternalApi paymentTransactionGatewayPostepayWebClient;
 
     @Autowired
-    private ObjectMapper objectMapper;
-
+    @Qualifier("paymentTransactionGatewayXPayWebClient")
+    XPayInternalApi paymentTransactionGatewayXPayWebClient;
 
-    public Mono<PostePayAuthResponseEntityDto> requestAuthorization(AuthorizationRequestData authorizationData) {
-        PostePayAuthRequestDto postePayAuthRequest = new PostePayAuthRequestDto()
-                .grandTotal(BigDecimal.valueOf(((long) authorizationData.transaction().getAmount().value()) + authorizationData.fee()))
-                .description(authorizationData.transaction().getDescription().value())
-                .paymentChannel(authorizationData.pspChannelCode())
-                .idTransaction(authorizationData.transaction().getTransactionId().value().toString());
+    @Autowired
+    private ObjectMapper objectMapper;
 
-        String encodedMdcFields = encodeMdcFields(authorizationData);
+    public Mono<Tuple2<Optional<PostePayAuthResponseEntityDto>,Optional<XPayAuthResponseEntityDto>>> requestGeneralAuthorization(AuthorizationRequestData authorizationData) {
```

Comment:
it's fine, maybe in the future we will understand how to model better (we had modeled like this here too https://github.com/pagopa/pagopa-ecommerce-transactions-service/blob/5ae77efa819464e7b0970f89dc83129218ae6f2e/src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java#L149)
we couldn't find a better way, right @giovanniberti ?

Reply:
Just an add . In this case method result is a Mono<Tuple2<Dto1,Dto2>> in place of Mono<Tuple2<Mono<Dto1>,Mono<Dto2>>>